client/Assets/Scripts/Battle/BattleManager.cs

```diff
@@ -114,6 +119,13 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
 					case Protobuf.Messages.Action.ActionTypeOneofCase.Death:
 						battleUnitsUI.Where(battleUnit => battleUnit.SelectedUnit != null).Single(unit => unit.SelectedUnit.id == action.Death.UnitId).DeathFeedback();
 						break;
+					case Protobuf.Messages.Action.ActionTypeOneofCase.EnergyRegen:
+						BattleUnit unit = battleUnitsUI.Single(unit => unit.SelectedUnit.id == action.EnergyRegen.TargetId);
```

Comment:
```suggestion
						BattleUnit unit = battleUnitsUI.Single(unit => battleUnit.SelectedUnit != null && unit.SelectedUnit.id == action.EnergyRegen.TargetId);
```

This change allows the battle to play out with teams with less than 6 untis, we should really change this "search" behavior and encapsulate it in a single method.